pkg/process/checks/process.go

```diff
@@ -137,6 +139,10 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo, oneShot bool
 	}
 	p.containerProvider = sharedContainerProvider
 
+	log.Info("Initializing gpu detector from process check")
+	p.gpuDetector = procutil.NewGPUDetector(p.wmeta)
```

Comment:
for now this is ok, but we'll want to use components and FX to inject this in instead of initializing this directly.

Reply:
Will address in #34137